cosine_similarity_model.py

```diff
@@ -0,0 +1,141 @@
+import joblib
+import logging
+import numpy as np
+from pymongo import MongoClient
+import pandas as pd
+from sklearn.metrics.pairwise import cosine_similarity
+from sklearn.feature_extraction.text import CountVectorizer
+
+
+class MovieRecommender:
+    def __init__(self):
+        self.logger = self.setup_logger()
+        self.all_movie_ids = []
+        self.all_movie_titles = {}
+        self.selected_movies = pd.DataFrame()
+
+
+    @staticmethod
+    def setup_logger():
+        logger = logging.getLogger('MovieRecommender')
+        logger.setLevel(logging.INFO)
+        handler = logging.StreamHandler()
+        formatter = logging.Formatter('%(asctime)s - %(levelname)s - %(message)s')
+        handler.setFormatter(formatter)
+        logger.addHandler(handler)
+        return logger
+
+
+    @staticmethod
+    def load_movies_from_db(movie_id=None):
+        client = MongoClient("mongodb://user:pass@localhost:27017")
+        db = client["movieDB"]
+        collection = db["movieDetails"]
+
+        query = {}
+        if movie_id:
+            query["_id"] = movie_id
+
+        projection = {
+            "_id": 1,
+            "title": 1,
+            "genres": 1,
+            "cast": 1,
+            "crew": 1,
+            "keywords": 1
+        }
+
+        movies = list(collection.find(query, projection))
+        return pd.DataFrame(movies)
+
+
+    def prepare_data(self, movies_df, liked_movie_ids):
+        self.all_movie_ids = movies_df["_id"].tolist()
+        self.all_movie_titles = movies_df.set_index("_id")["title"].to_dict()
+        self.selected_movies = movies_df[movies_df["_id"].isin(liked_movie_ids)]
+
+
+    @staticmethod
+    def create_soup(movie):
```

Comment:
approach using `create_soup` in `prepare_vectors` is valid, but column wise processing offers better performance on large datasets
you can simplify it by applying transformations directly to columns, like this 
```
genres = movies_df['genres'].apply(lambda x: [i['name'] for i in x])
cast = movies_df['cast'].apply(lambda x: [i['name'] for i in x] if isinstance(x, list) else [])
```
combine the processed columns into the soup
`soup = cast + genres + ...`
this approach also eliminates string concatenation at each step